app/src/main/java/com/rasteplads/festfriend/FestFriendApplication.kt

```diff
@@ -19,13 +33,46 @@ import com.rasteplads.festfriend.pages.MapPage
 import com.rasteplads.festfriend.ui.theme.FestFriendTheme
 import kotlinx.coroutines.delay
 
+enum class JoinOrCreate { Join, Create, None }
+
 @Composable
 fun FestFriendApplication(appViewModel: AppViewModel = viewModel()){
 
     val appState by appViewModel.uiState.collectAsState()
     val navController = rememberNavController()
-    val navToMap = {navController.navigate(FestFriendScreen.Map.name)}
-    FestFriendTheme (dynamicColor = false){
+    val navToMap = { navController.navigate(FestFriendScreen.Map.name) }
+    val context = LocalContext.current
+    val locationClient = remember {
+        LocationServices.getFusedLocationProviderClient(context)
+    }
+    var action = JoinOrCreate.None
+    val permission = Manifest.permission.ACCESS_FINE_LOCATION
+    val locationPermissionLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted ->
+        if (isGranted) {
```

Comment:
Der er rimelig meget nesting her. brug guard clauses is stedet for:

if (!isGranted){
val builder....
....
return
}

success kode